File: src/LCR101.NUPfPr.1000287/solution.java
class Solution {
    public boolean canPartition(int[] nums) {
        int sum = 0;
        for(int num : nums){
            sum += num;
        }
        if(sum%2 == 1){
            return false;
        }
        int half = sum>>1;
        boolean[] cans = new boolean[half+1];
        cans[0] = true;
        for(int num : nums){
            if(num>half){
                return false;
            }
            for(int i=half;i-num>-1;i--){
                cans[i] = cans[i] || cans[i-num];
            }
        }
        
        return cans[half];
    }
}

File: src/LCR101.NUPfPr.1000287/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canPartition = function (nums) {
    const sum = nums.reduce((sum, item) => {
        return sum + item;
    }, 0);
    if (sum % 2 === 1) {
        return false;
    }
    const half = sum >> 1;
    const dp = new Array(half + 1).fill(false);
    dp[0] = true;
    for (let i = 0; i < nums.length; i++) {
        for (let j = dp.length - 1; j - nums[i] > -1; j--) {
            dp[j] = dp[j] || dp[j - nums[i]];
        }
    }
    return dp[half];
};
